src/random.h

```diff
@@ -67,7 +67,17 @@
  * Thread-safe.
  */
 void GetRandBytes(unsigned char* buf, int num) noexcept;
+/** Generate a random integer in the range [0..range). Precondition: range > 0 */
 uint64_t GetRand(uint64_t nMax) noexcept;
+// The following GetRand*(std::chrono::*) MUST NOT be templated on the type of
+// the duration to prevent unsafe use at the call site. If it was templated, it
```

Comment:
My knowledge is pretty limited, but maybe I can make a suggestion for clearer wording?

It seems like the issue is that, if `GetRand*(std::chrono::*)` is templated, the `max_duration`'s count() is used as-is instead of converting between the argument type and return type like we would expect it to. Instead, the conversion happens at the end. In simple terms, the `std::chrono` types won't convert nicely which means we lose precision if we want random small units of time over the duration of a large unit of time (which I'm guessing is a common use case). For example, when you try to get random seconds over the duration of 1 hour, the only return value you can get is 0 hours = 0 seconds.

Other example: `std::chrono::seconds rand_seconds = GetRandDur(std::chrono::hours{4});` can return non-zero values, but only at the precision of hours (i.e. 0, 3600, 7200, or 10800 seconds). 

If you were to make a template, it only necessarily works as intended if you pass in a `max_duration` of strictly the same type (or @sipa's workaround).

Reply:
@MarcoFalke- I built on @gzhao408's description. here's an attempt at a comment: 

The following `GetRand<duration>` MUST NOT be templated on the type of the `<duration>` to prevent unexpected results. If there is a mismatch between the return type and the argument type, the compiler is unable to reliably select the correct template. The unexpected conversion handling can lead to precision loss. 

Eg. 
`std::chrono::seconds rand_seconds = GetRandDur(std::chrono::hours{2})` will return 1 of 2 values rather than 1 of 120 values.

Reply:
Thanks @gzhao408 and @amitiuttarwar for helping me to explain this. Took your suggestion and added you both as co author in the commit.